lib/sqewer/local_connection.rb

```diff
@@ -0,0 +1,134 @@
+class Sqewer::LocalConnection < Sqewer::Connection
+  FAIL_AFTER_DELIVERIES = 10
+
+  def with_db(**k)
+    SQLite3::Database.open('sqewer-local-queue.sqlite3', **k) do |db|
+      db.busy_timeout = 5
+      return yield db
+    end
+  end
+
+  def with_readonly_db(&blk)
+    with_db(readonly: true, &blk)
+  end
+
+  def initialize(queue_url)
+    require 'sqlite3'
+    @queue_url = queue_url
+    with_db do |db|
+      db.execute("CREATE TABLE IF NOT EXISTS sqewer_messages_v1 (
+        id INTEGER PRIMARY KEY AUTOINCREMENT ,
+        queue_url VARCHAR NOT NULL,
+        receipt_handle VARCHAR NOT NULL,
+        deliver_after_epoch INTEGER,
+        times_delivered_so_far INTEGER DEFAULT 0,
+        last_delivery_at_epoch INTEGER,
+        visible BOOLEAN DEFAULT 't',
+        message_body TEXT)"
+      )
+      db.execute("CREATE INDEX IF NOT EXISTS index_sqewer_messages_v1_on_receipt_handle ON sqewer_messages_v1 (receipt_handle)")
+      db.execute("CREATE INDEX IF NOT EXISTS index_sqewer_messages_v1_on_queue_url ON sqewer_messages_v1 (queue_url)")
+    end
+  rescue LoadError => e
```

Comment:
Why do we push this onto the user instead of putting it in the Sqewer gemfile? It's a large dependency, sure, but this seems rather inconsiderate.

Reply:
This is also what ActiveRecord does for the database drivers. As long as we say explicitly what the user has to do to make this work - what's the issue? I.e. see comment on peer dependencies above.